smb1pdu.c

```diff
@@ -5877,7 +5877,11 @@ static int smb_populate_readdir_entry(struct ksmbd_conn *conn, int info_level,
  *
  * Return:	0 on success, otherwise -EINVAL
  */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+static bool ksmbd_fill_dirent(struct dir_context *ctx, const char *name, int namlen,
```

Comment:
I'm sorry, but wouldn't this change contradicts with the semantic requirements of the new calling conventions of `filldir_t`? The new calling conventions now requires returning `true` for those case that previously return `0`, and `false` for those returning `-E`.

See https://github.com/torvalds/linux/commit/25885a35a72007cf28ec5f9ba7169c5c798f7167#diff-c031ce4f31310be3aad85fc256d71d769cb64808a53c5160c588984fba6487bcR925 for details